src/types/type_wrappers.rs

```diff
@@ -0,0 +1,81 @@
+use crate::utils::js_error;
+
+#[napi]
+pub enum CqlType {
+    Ascii,
+    Boolean,
+    Blob,
+    Counter,
+    Decimal,
+    Date,
+    Double,
+    Duration,
+    Empty,
+    Float,
+    Int,
+    BigInt,
+    Text,
+    Timestamp,
+    Inet,
+    List,
+    Map,
+    Set,
+    UserDefinedType,
+    SmallInt,
+    TinyInt,
+    Time,
+    Timeuuid,
+    Tuple,
+    Uuid,
+    Varint,
+    Custom,
+}
+
+#[napi]
+#[derive(Clone)]
+pub struct ComplexType {
+    pub base_type: CqlType,
+    pub(crate) support_type_1: Option<Box<ComplexType>>,
+    pub(crate) support_type_2: Option<Box<ComplexType>>,
+}
```

Comment:
:wrench: This does allow for erroneous variants:
1. `ComplexType { base_type: Uuid, support_type_1: None, support_type_2: Some(...) }`
2. `ComplexType { base_type: Tuple, support_type_1: None, support_type_2: None }`

`ComplexType` would be a good representation when and only when it would permit creation of correct instantiations and would prevent creation of incorrect ones.